consumer-test/runTests.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
+
+LIBRARY_VERSION_UNDER_TEST=$(git log --format="%H" -n 1)
+REPOSITORY_UNDER_TEST=${TRAVIS_REPO_SLUG:-SAP/jenkins-library}
+
+rm -rf workspace
+git clone -b consumer-test https://github.com/sap/cloud-s4-sdk-book workspace
+cp -f jenkins.yml workspace
+cd workspace
+sed -i -e "s:__REPO_SLUG__:${REPOSITORY_UNDER_TEST}:g" jenkins.yml
+echo "@Library(\"piper-library-os@$LIBRARY_VERSION_UNDER_TEST\") _" | cat - Jenkinsfile > temp && mv temp Jenkinsfile
+git commit --all --author="piper-testing-bot <null@null.com>" --message="Set piper lib version for test"
+
+docker run -v /var/run/docker.sock:/var/run/docker.sock -v "${PWD}":/workspace -v /tmp -e CASC_JENKINS_CONFIG=/workspace/jenkins.yml -e CX_INFRA_IT_CF_USERNAME -e CX_INFRA_IT_CF_PASSWORD -e BRANCH_NAME=consumer-test ppiper/jenkinsfile-runner
```

Comment:
we execute pipeline script ` cloud-s4-sdk-book/Jenkinsfile` from branch `consumer-test` right?

This file in turn materialized one other repo and hands over execution to that. What is the reason for having the indirection?

Reply:
`cloud-s4-sdk-book` is the place for our canonical example project. I _could_ also copy the code in here, but I don't think we want this, as it will make it harder to maintain it in sync. Also, this would add many files here to the lib which are not part of the lib.

As always, I'm open for suggestions.